### ISSUE TITLE
Cannot view image from private asset container

### ISSUE THREAD

**Issue by morhi** [statamic/cms#7552]
### Bug description

According to the [official docs](https://statamic.dev/assets/#private-containers) I can create a private asset container to prevent files from being accessible publicly. I have done this before, but the feature seems to be broken. Everything seems to work fine, except these two things:

1. Opening an (image) asset won't show the image, because of an exception thrown: `Invalid preset`. This happens, because the image URL does not contain a preset at the end of the URL, like `small`. However, when I append `small` manually to the URL the image is loaded as expected.
2. Even though the filesystem behind the container does not have a `url` set, the asset context menus still provide an URL to the asset, which does not exist obviously. 

The exception is this:

```
Exception: Invalid preset in file .../vendor/statamic/cms/src/Http/Controllers/CP/Assets/ThumbnailController.php on line 110
```

Some screenshots:
![Screenshot 2023-02-16 at 10 35 59](https://user-images.githubusercontent.com/12037294/219326459-32118176-baac-41cd-acfe-631649d7e126.png)

![Screenshot 2023-02-16 at 10 48 56](https://user-images.githubusercontent.com/12037294/219329955-377b0440-8fa2-4ed1-91d9-53617601d0a9.png)

### How to reproduce

1. Create a new asset container linked to a private Laravel filesystem like this:

```php
       'private' => [
            'driver' => 'local',
            'root' => storage_path('app/private'),
            'visibility' => 'private',
        ],
```

2. Upload an image to the container
3. Open the new image from the CP
4. See the error

It can be fixed by setting an `url` property and providing a `root` path that is publicly available.

### Environment

```yaml
Environment
Laravel Version: 9.51.0
PHP Version: 8.1.7
Composer Version: 2.2.5
Environment: local
Debug Mode: ENABLED
URL: localhost
Maintenance Mode: OFF

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Drivers
Broadcasting: log
Cache: statamic
Database: mysql
Logs: stack / single
Mail: smtp
Queue: sync
Session: file
```

### Installation

Fresh statamic/statamic site via CLI

### Antlers Parser

runtime (new)

**Comment by phyroslam** [comment]
I came across this issue recently and I have traced the failure to the CP Asset Resource class where the previewUrl() method is not passing a valid preset, I fixed by passing the 'small' preset since apparently those are the only ones available in the Imaging manager class. Certainly it needs to be fixed in the source code.

![Screen Shot 2023-02-21 at 10 10 49 AM](https://user-images.githubusercontent.com/5395518/220425734-0d09b16b-224f-40c9-b2a4-76be8e2255fd.png)

![Screen Shot 2023-02-21 at 10 14 02 AM](https://user-images.githubusercontent.com/5395518/220426401-526d504b-9c28-473d-9c6c-b50451436b6d.png)

**Comment by henzeb** [comment]
I have a similar issue. If I open the image url in a seperate tab, I see a black screen with a white square in the middle.

in the ResponseFactory, adding the line `if (ob_get_level()) ob_end_clean();` just before returning the response seems to fix it. 
I don't get any other error messages, just a black square.
It also doesn´t have anything to do with file generation, because the file in the cache looks good to me. It has something to do with the output buffer. But I haven't figured out what.

**Comment by jasonvarga** [comment]
The thumbnail error was fixed in #7664.

The copy action url will be hidden for private assets in #8005

If you have other issues, please open them separately.
